api/types/container/network.go

```diff
@@ -0,0 +1,339 @@
+package container
+
+import (
+	"errors"
+	"fmt"
+	"iter"
+	"strconv"
+	"strings"
+	"unique"
+)
+
+// PortProto represents a network protocol for a port.
+type PortProto string
```

Comment:
`PortProto` sounds like a weird name to me. Every time I see it I instinctively think that it's a union of a port number and a proto, whereas it's just a proto. Maybe we can rename that type into `Proto` 🤔